rotkehlchen/chain/evm/decoding/curve_lend/decoder.py

```diff
@@ -617,14 +632,102 @@ def _decode_vault_controller_events(self, context: DecoderContext) -> DecodingOu
 
         return DEFAULT_DECODING_OUTPUT
 
+    def _decode_staking_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] not in (GAUGE_DEPOSIT, CURVE_VAULT_GAUGE_WITHDRAW):
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[:32]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        gauge_asset = self.base.get_or_create_evm_asset(context.tx_log.address)
+        paired_events_data, from_event_type, from_event_subtype, to_event_type, to_event_subtype, to_notes = None, None, None, None, None, ''  # noqa: E501
+        for event in context.decoded_events:
+            if (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == amount and
+                    event.address == context.tx_log.address
+            ):
+                event.counterparty = CPT_CURVE
+                event.product = EvmProduct.GAUGE
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_FOR_WRAPPED
+                event.notes = f'Deposit {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} into {gauge_asset.symbol}'  # noqa: E501
+
+                paired_events_data = ([event], True)
+                from_event_type = HistoryEventType.RECEIVE
+                from_event_subtype = HistoryEventSubType.NONE
+                to_event_type = HistoryEventType.RECEIVE
+                to_event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                to_notes = f'Receive {amount} {gauge_asset.symbol} after depositing in curve lending vault gauge'  # noqa: E501
+
+            elif (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == amount and
+                    event.address == context.tx_log.address
+            ):
+                event.counterparty = CPT_CURVE
+                event.product = EvmProduct.GAUGE
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                event.notes = f'Withdraw {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from {gauge_asset.symbol}'  # noqa: E501
+
+                paired_events_data = ([event], False)
+                from_event_type = HistoryEventType.SPEND
+                from_event_subtype = HistoryEventSubType.NONE
+                to_event_type = HistoryEventType.SPEND
+                to_event_subtype = HistoryEventSubType.RETURN_WRAPPED
```

Comment:
not sure how this works. This function decodes both deposit and withdrawal right? State it in the docs then. Also you can break in each of the branches since you need one and not both branches executed